stdlib/private/SwiftPrivateLibcExtras/SwiftPrivateLibcExtras.swift

```diff
@@ -125,6 +125,8 @@ public func _stdlib_pipe() -> (readEnd: CInt, writeEnd: CInt, error: CInt) {
   let ret = fds.withUnsafeMutableBufferPointer { unsafeFds -> CInt in
 #if os(Windows)
     return _pipe(unsafeFds.baseAddress, 0, 0)
+#elseif os(WASI)
+    preconditionFailure("no pipes available on WebAssembly/WASI")
```

Comment:
IIRC, this is largely used for the reflection tests.  Perhaps we should just disable the reflection tests instead.  I really think that given the amount of stuff not available in WASI libc yet, the focus should be improving that first (which is why I stopped and started looking at WASI libc instead).

Reply:
This is used in `RaceTest.swift` in the implementation of `_InterruptibleSleep` on non-Windows platforms. Would `_InterruptibleSleep` need a dedicated implementation for WASI instead? Could WebAssembly/WASI#77 be somewhat related?